integration_tests/src/main/python/conftest.py

```diff
@@ -77,6 +77,15 @@ def is_emr_runtime():
 def is_dataproc_runtime():
     return runtime_env() == "dataproc"
 
+def get_test_tz():
```

Comment:
Q: where would this func by used? also pytest conf would relying on run_pyspark script seems weird

can you at least try `os.environ.get('PYSP_TEST_spark_sql_session_timeZone', 'UTC')` to make sure it could have a default?

Reply:
these funcs will be used in pytest xfail. For an existing test case before operator supports non-utc: add xfail(is_non_utc()).

>also pytest conf would relying on run_pyspark script seems weird.

I tested we can get this config in the `conftest`.
conftest is internal file of IT, so it's safe to get Env variable from run_pyspark.

Reply:
> conftest is internal file of IT, so it's safe to get Env variable from run_pyspark.

hmm OK, for me pytest code itself should at least provide the defaults here

I would like to hear more feedback from other developers~

Reply:
If does not set it, then an error throws:

```
>>> os.environ["PYSP_TEST_spark_sql_session_timeZone"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<frozen os>", line 679, in __getitem__
KeyError: 'PYSP_TEST_spark_sql_session_timeZone'
```

This error will force us to set this cfg.

Reply:
I just threw the code together so a get with a default looks like a great addition for safety/robustness.

Reply:
Agree with @pxLi, this seems weird.  There's already a standard environment variable for users to convey what timezone they want an application to use, and that's `TZ` (see environ(7) manpage).  Is there a reason to avoid using that when running the tests?  Then the user can simply set TZ when they run the tests to specify the timezone to use (e.g.: TZ=CST integration_tests/run_pyspark_from_build.sh ...).  I already do this when running Spark in local mode to get it to use the timezone desired, and it would be nice to avoid adding yet another way to specify a timezone that isn't compatible with the standard way to do it.

Reply:
Changed from TEST_TZ to TZ.
Note:  can not get the TZ env variable in conftest.py. So still use  `PYSP_TEST_spark_sql_session_timeZone` to pass time zone.

Reply:
> can not get the TZ env variable in conftest.py.

Can you elaborate on this?  I'm confused how we can export one variable and read it in conftest but somehow can't do the same to another.  Is something in the shell startup environment bashing the TZ variable?

Reply:
@res-life wrapping ```TZ``` with ```PYSP_TEST_spark_sql_session_timeZone``` wouldn't be same as original way we did in [is_tz_utc](https://github.com/NVIDIA/spark-rapids/pull/9652/files#diff-0271a6b980cb547f3b0f7a18ab9109357512c6c3562fd3631cbceae3bd3542f4L58)?

Java systemDefault will respect TZ environment.
```java 
import java.time.ZoneId;

public class Test{
   public static void main(String [] args) {
     System.out.println("time zone is " + ZoneId.systemDefault());
   }

}
```
```
$ export TZ="UTC" && java Test
time zone is UTC
$ export TZ="Asia/Shanghai" && java Test
time zone is Asia/Shanghai
```

Reply:
Offline discussed with @res-life , let's add comments that we need to get utc time before spark session starts.

Reply:
We can not use the following code when the Spark session is not started.
```python
jvm = spark.sparkContext._jvm
    utc = jvm.java.time.ZoneId.of('UTC').normalized()
    sys_tz = jvm.java.time.ZoneId.systemDefault().normalized()
```

Reply:
>can not get the TZ env variable in conftest.py.

Because TZ is rewrite to UTC in [data_gen.py](https://github.com/NVIDIA/spark-rapids/blob/v23.10.0/integration_tests/src/main/python/data_gen.py#L33-L36)
```
os.environ['TZ'] = 'UTC'
time.tzset()
```

Removed the above code. See my last commit:
I use Env variable TZ now.
And I updated TimestampGen to avoid generate out of range timestamp